src/braket/analog_hamiltonian_simulator/rydberg/validators/atom_arrangement.py

```diff
@@ -86,3 +94,22 @@ def filling_same_length_as_sites(cls, values):
                 f"Filling length ({length}) does not match sites length ({expected_length})"
             )
         return values
+
+    # Two lattice sites cannot be closer (in terms of Euclidean distance) than MIN_DISTANCE
+    @root_validator(pre=True, skip_on_failure=True)
+    def sites_not_too_close(cls, values):
+        sites = values["sites"]
+        capabilities = values["capabilities"]
+        for idx1, s1 in enumerate(sites):
+            for idx2, s2 in enumerate(sites):
```

Comment:
```suggestion
            for index_2, site_2 in enumerate(sites[index_1 + 1:], start=index_1 + 1):
```
then you can get rid of the check below